docs: add WASM config for Apache and Nginx (#4246)
unoplatform/uno: doc/articles/getting-started/deployments.md

> # WASM Web Server Configuration

Reviewer: ```suggestion
# How to host a WebAssembly App
```

- [WASM Web Server Configuration](#wasm-web-server-configuration)
  - [Nginx](#nginx)
  - [Apache](#apache)

> Regardless of what webserver (reverse proxy) you are using, you will at core need to support the following Content (MIME) types.

Reviewer: ```suggestion
Regardless of the webserver (or reverse proxy) software used, the support the following Content (MIME) types are always needed:
```

-   `application/wasm`
-   `application/octet-stream`

> -   `application/octet-stream`

Reviewer: Duplicate ?

Author: Yes, I have removed the duplicate.

-   `application/font-woff`

> -   `application/font-woff`

Reviewer: Duplicate ?

## Nginx

> Below contains a extremly performance tuned version of *nginx* fitting WASM deployments. The MIME types have been extended on top of the base *nginx* MIME types with WASM support.

Reviewer: ```suggestion
Below contains a performance tuned version of *nginx* fitting WebAssembly deployments. The MIME types have been extended on top of the base *nginx* MIME types with WASM support.
```

  worker_connections 4000;

  # optmized to serve many clients with each thread, essential for linux -- for testing environment
  use epoll;

  # accept as many connections as possible, may flood worker connections if set too low -- for testing environment
  multi_accept on;
}

http {
  include /etc/nginx/mime.types;
  default_type application/octet-stream;

  types {
    application/wasm wasm;
    application/octet-stream clr;
    application/octet-stream pdb;
    application/font-woff woff;
    application/font-woff woff2;
  }

  log_format main '$remote_addr - $remote_user [$time_local] "$request" '
  '$status $body_bytes_sent "$http_referer" '
  '"$http_user_agent" "$http_x_forwarded_for"';

  # cache informations about FDs, frequently accessed files
  # can boost performance, but you need to test those values
  open_file_cache max=200000 inactive=20s;
  open_file_cache_valid 30s;
  open_file_cache_min_uses 2;
  open_file_cache_errors on;

  # to boost I/O on HDD we can disable access logs
  access_log off;

  # copies data between one FD and other from within the kernel
  # faster then read() + write()
  sendfile on;

>   # send headers in one peace, its better then sending them one by one

Reviewer: ```suggestion
  # Send headers in one piece, it's better then sending them one by one
```

  tcp_nopush on;

  # don't buffer data sent, good for small data bursts in real time
  tcp_nodelay on;

>   # reduce the data that needs to be sent over network -- for testing environment

Reviewer: Why is this test environment only ? Wasm apps benefit a lot from compression, and even better if brotli can be enabed.

Author: Added Brotli & removed test environment comment.

  gzip on;
  gzip_min_length 10240;
  gzip_proxied expired no-cache no-store private auth;
  gzip_types text/plain text/css text/xml text/javascript application/x-javascript application/json application/xml;
  gzip_disable msie6;

  # allow the server to close connection on non responding client, this will free up memory
  reset_timedout_connection on;

  # request timed out -- default 60
  client_body_timeout 10;

  # if client stop responding, free up memory -- default 60
  send_timeout 2;

  # server will close connection after this time -- default 75
  keepalive_timeout 30;

  # number of requests client can make over keep-alive -- for testing environment
  keepalive_requests 100000;

  #########################################
  # Just For Security Reason
  #########################################

  # Security reasons, turn off nginx versions
  server_tokens off;

>   more_clear_headers Server; # Custom module: headers-more-nginx-module (https://github.com/openresty/headers-more-nginx-module)

Reviewer: Is this a default configuration or do users need to install something additional ?

Author: Removed this as it requires an additional module.
